cmd/internal/flowmode/cmd_run.go

```diff
@@ -383,6 +384,10 @@ func loadFlowSource(path string, converterSourceFormat string, converterBypassEr
 			}
 
 			bb, err := os.ReadFile(curPath)
+			// Retain the original error and don't try to convert if an error occurred.
+			if err != nil {
+				bb, err = config.EnsureUTF8(bb)
+			}
```

Comment:
Technically this is a spec change for River files: https://grafana.com/docs/agent/latest/flow/config-language/files/

> River files are required to be UTF-8 encoded, and are permitted to contain Unicode characters. River files can use both Unix-style line endings (LF) and Windows-style line endings (CRLF), but formatters may replace all line endings with Unix-style ones.

Reply:
How would you feel about adding a check that if the config file is not utf8bom/nobom, we return an error stating that it needs to be utf8? Right now if you pass in an utf16 you get a a slew of invalid character errors.

Reply:
Sure, I think that would be OK

Reply:
Error on startup is now similar to `Error: reading config path "/home/mdurham/utils/flow/rw.river": configuration is encoded with UTF16LittleEndian but must be utf8`